src/test/java/productDependency/CreateCategoryTest.java

```diff
@@ -0,0 +1,87 @@
+package productDependency;
+
+import baseClassPackage.BaseClasses;
+import java.io.IOException;
+import org.openqa.selenium.WebDriver;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import utilPackages.PropertyValExtractors;
+import productDependency.CreateCategoryPage;
+
+
+
+public class CreateCategoryTest {
+	CreateCategoryPage categoryPage;
+    WebDriver driver,driver2;
+
+    @BeforeClass(alwaysRun=true)
+    public void setup(){
+        driver= BaseClasses.setup( "User1");
+        PropertyValExtractors p=new PropertyValExtractors();
+        p.getPropertyFile("test", "configuration.properties");
+        String url=p.getVal("url1");
+        driver.get(url);
+        //WaitClass.waitForUrl(driver, url, 10);
+        categoryPage=CreateCategoryPage.getPage(driver, CreateCategoryPage.class);
+
+    }
+
+    @Test(groups={"Product Dependency Feature"},description="Add Edit Category")
+    public void userSignin() throws IOException{
+
+    	/*categoryPage.enterLoginID();
+    	categoryPage.enterPassword();
+    	categoryPage.selectCompany();
+    	categoryPage.clickLoginButton();*/
+    	
+    	categoryPage.clickProductsTab();
+    	categoryPage.clickAddCategoryButton();
+    	categoryPage.enterCategoryName();
+    	categoryPage.selectCompanies();
+    	categoryPage.clickSaveChangesButton();
+    	categoryPage.verifyConfirmationMsg();
+    	
+    	categoryPage.clickProductsTab();
+    	categoryPage.selectCategoryName();
+    	categoryPage.clickAddProductButton();
+    	categoryPage.enterEnglishDescription();
+    	categoryPage.enterProductCode();
+    	categoryPage.selectCompany1();
+    	categoryPage.selectPriceCurrency();
+    	categoryPage.enterPriceRate();
+    	categoryPage.clickAddPriceButton();
+    	categoryPage.clickSaveChangesButton();
+        categoryPage.verifyConfirmationMsg();
+        
+        
+        
+        categoryPage.clickProductsTab();
+    	categoryPage.selectCategoryName();
```

Comment:
@mayankchhajer Format the code by using Ctrl+Shift+F , and maintain the gaps properly by giving the Product name in the comment

Reply:
Done